menas/src/main/scala/za/co/absa/enceladus/menas/controllers/PropertyDefinitionController.scala

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.menas.controllers
+
+import java.net.URI
+import java.util.concurrent.CompletableFuture
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.{HttpStatus, ResponseEntity}
+import org.springframework.security.core.annotation.AuthenticationPrincipal
+import org.springframework.security.core.userdetails.UserDetails
+import org.springframework.web.bind.annotation._
+import za.co.absa.enceladus.menas.services.PropertyDefinitionService
+import za.co.absa.enceladus.model.properties.PropertyDefinition
+
+import scala.concurrent.ExecutionContext.Implicits.global
+
+@RestController
+@RequestMapping(path = Array("/api/properties/datasets"), produces = Array("application/json"))
+class PropertyDefinitionController @Autowired()(propertyDefService: PropertyDefinitionService)
+  extends VersionedModelController(propertyDefService) {
+
+  import za.co.absa.enceladus.menas.utils.implicits._
+
+  @GetMapping(Array(""))
+  def getAllDatasetProperties(): CompletableFuture[Seq[PropertyDefinition]] = {
```

Comment:
I think we should try to avoid top-level primitives wherever possible in our API. Also by the definition, an object should always be the top-level entity. It helps with extensibility as well. Imagine you want to add some extra metadata to this response. With Array as a top-level object, you would have to rewrite a lot of thing in a lot of places.

Some links:
- https://www.json.org/json-en.html
- https://security.stackexchange.com/questions/7001/how-should-web-app-developers-defend-against-json-hijacking/7003#7003

Still, I am aware of the new standard https://www.rfc-editor.org/rfc/rfc7159.txt saying that JSON can start with an array. But in addition to objects and arrays, it can now also be a number, string or the values false, null or true. And even though I know this, I would have an object first.

Reply:
Are you objecting against the top-level-primitives usage only? I.e. wrapping it with an object and be done with it? Sure, that can be done.

Or are you concerned with returning this much data at-once, too? (I think we need a way to list the properties, but we can do it using `/api/properties/dataset/list` in order to get a "versioning summary" only, that the caller might use to retrieve full objects individually - this depends on the intended usecase, though.)

Reply:
Top-level primitives and arrays only. As you say, just wrap it in an object.

Reply:
Agreed in general, but out of scope for this PR to properly solve all occurrences throughout the project. Creating a separate issue: https://github.com/AbsaOSS/enceladus/issues/1616